File: adapters/src/main/java/com/wang/container/bean/IContainerBean.java
package com.wang.container.bean;

import androidx.annotation.NonNull;

import com.wang.container.adapter.IContainerItemAdapter;

/**
 * 你的最外层bean必须继承该接口
 */
public interface IContainerBean {
    /**
     * 这个bean属于哪个adapter
     */
    @NonNull
    Class<? extends IContainerItemAdapter> getBindAdapterClass();
}

File: adapters/src/main/java/com/wang/container/bean/IContainerBean.kt
package com.wang.container.bean

import com.wang.container.adapter.IContainerItemAdapter

/**
 * 你的最外层bean必须继承该接口
 */
interface IContainerBean {
    /**
     * 这个bean属于哪个adapter
     */
    fun getBindAdapterClass(): Class<out IContainerItemAdapter<*>>
}